### PR TITLE
Update dependencies to fix safety check failures

### PR DESCRIPTION
**Pull request by travis-newby**
Update dependencies causing safety check to fail. There were two dependencies updated:

1. Pillow was updated to a version that is not [susceptible to DoS](https://pyup.io/v/51886/f17) (9.3.0). Pillow was previously pinned, so I left it as a pinned dependency; there are _upcoming_ breaking changes specified in the release notes, so this seems like a safe choice. There are some new deprecations, but I didn't see any _current_ breaking changes in the release notes. 
2. Certifi was updated to a version that [removes TrustCor certificates](https://pyup.io/v/52365/f17/) from the root store. (>= 2022.12.07).

During this process, setuptools was upgraded automatically. That takes care of a potential [DoS issue](https://pyup.io/v/52495/f17).

Finally, I removed the nb-black dev dependency. It's not being used, and it doesn't play nice with poetry on certain Mac devices.

### CODE REVIEW

**Comment by travis-newby** [comment]
I'd like to do a bit more testing before merging, but I first wanted to see if this resolved issues we saw in the checks.

File: data/data-pipeline/data_pipeline/ipython/compare_tiles_and_geoJson_files.ipynb [hunk 1]
```diff
@@ -332,7 +331,7 @@
  ],
  "metadata": {
   "kernelspec": {
-   "display_name": "Python 3.8.10 ('data-pipeline-WziHKidv-py3.8')",
+   "display_name": "Python 3 (ipykernel)",
```
Comment (vim-usds) [inline data/data-pipeline/data_pipeline/ipython/compare_tiles_and_geoJson_files.ipynb:334]: I'm a noob with notebooks, what is the significance of the display name and do we know when it's used?

Comment (vim-usds) [reply data/data-pipeline/data_pipeline/ipython/compare_tiles_and_geoJson_files.ipynb:334]: Any reason to not include the semantic versioning?

**Review by lucasmbrown-usds — APPROVED** [review]
LGTM!

**Comment by travis-newby** [comment]
Spot checked around the map. Everything looks good. 🚀

**Pull request merged**